packages/plugin-github/sources/githubUtils.ts

```diff
@@ -1,12 +1,5 @@
-const githubPatterns = [
-  /^github:([^\/#]+)\/([^\/#]+)(?:#(.+))?$/,
-  /^git@github.com:([^\/#]+)\/([^\/#]+)(?:#(.+))?$/,
-  /^git\+https:\/\/github.com\/([^\/#]+)\/([^\/#]+)\.git?(?:#(.+))?$/,
-  /^https:\/\/github.com\/([^\/#]+)\/([^\/#]+)\/tarball\/([^\/#]+)(?:#|$)/,
-];
-
 export function isGithubUrl(string: string) {
-  return githubPatterns.some(pattern => !!string.match(pattern));
```

Comment:
I understand that it's more optimized, but I would prefer to keep the various patterns clearly splitted. Regexps are hard to maintain, and I want to make it clear to external users which patterns are used and which ones aren't.

Also note that this function is meant to return packages that are specific to GitHub, not all Git repositories (cf the main comment).

Reply:
I, too, arrived at this after seeing how it difficult it would be write a regex and capture the needed groups.